activerecord/lib/active_record/relation.rb

```diff
@@ -642,6 +642,13 @@ def inspect
       "#<#{self.class.name} [#{entries.join(', ')}]>"
     end
 
+    def empty_scope? # :nodoc:
+      @values.empty? ||
```

Comment:
I intended to avoid extra `WhereClause.new`, `Relation.create`, and `@values.dup`.
Do you prefer `values == klass.unscoped.values`?